CRM/CiviAwards/Enable/ActivateCustomGroupSupportForAwardsCategory.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+use CRM_CiviAwards_Helper_CaseTypeCategory as CaseTypeCategoryHelper;
+use CRM_Civicase_Service_CaseCategoryCustomFieldExtends as CaseCategoryCustomFieldExtends;
+
+/**
+ * Class ActivateCustomGroupSupportForAwardsCategory.
+ */
+class CRM_CiviAwards_Enable_ActivateCustomGroupSupportForAwardsCategory {
+
+  /**
+   * Deletes the Awards option from the CG Extends option values.
+   */
+  public function apply() {
+    $caseCategoryCustomFieldExtends = new CaseCategoryCustomFieldExtends();
+    $caseCategoryCustomFieldExtends->activate(CaseTypeCategoryHelper::AWARDS_CASE_TYPE_CATEGORY_NAME);
+  }
```

Comment:
We need to approach this a bit differently. The issue is that the users can add more categories of type awards via the UI apart from those added via extension installation and each of these award categories will have corresponding option value entries in the `cg_extends` option group. 
We should probably do something like this:

```php
  public function apply() {
    $caseCategoryDetails = array_column(CRM_Civicase_Helper_CaseCategory::getCaseCategories(), 'name', 'value');
    $applicantCaseCategoryIds = CRM_CiviAwards_Helper_CaseTypeCategory::getApplicantManagementCaseCategories();
    $caseCategoryCustomFieldExtends = new CaseCategoryCustomFieldExtends();
    foreach ($applicantCaseCategoryIds as $caseCategoryValue) {
      $caseCategoryCustomFieldExtends->activate($caseCategoryDetails[$caseCategoryValue]);
    }
  }
```

we should do same for the deactivation too.

Reply:
Good catch!
Updated the PR to activated/deactivated any custom groups created by both the extension or the user.

Reply:
Looks like you misunderstood what I meant with this comment. We don't need any change in the civicase PR to accomplish what I said. Basically this class only takes care of the Awards case category which is installed in the extension by default and we need to do this too for any other case categories created by the user. Your initial code should work fine, we only need to do a loop to do same thing for the other case categories. the code I pasted earlier is still valid
```php
  public function apply() {
    $caseCategoryDetails = array_column(CRM_Civicase_Helper_CaseCategory::getCaseCategories(), 'name', 'value');
    $applicantCaseCategoryIds = CRM_CiviAwards_Helper_CaseTypeCategory::getApplicantManagementCaseCategories();
    $caseCategoryCustomFieldExtends = new CaseCategoryCustomFieldExtends();
    foreach ($applicantCaseCategoryIds as $caseCategoryValue) {
      $caseCategoryCustomFieldExtends->activate($caseCategoryDetails[$caseCategoryValue]);
    }
  }
```
If you can undo changes done in civicase and use this code, should work just fine.